### PR TITLE
UISP-11: Build BigTest unit test code coverage above 80%

### PR DESCRIPTION
**Pull request by K-Felk**
This update adds automated test coverage for this module.
there are two tests, one for each modal the handler displays.
the first test (accessModal) loads a dummy stripes-core instance
and generates a routing event to confirm that ui-sp will intercept
the event and generate a warning modal. The second test just
mounts the servicepoint selection modal and checks to make sure
it displays servicepoint data correctly. Actually changing the
users default service point is handled by stripes-core, and is
not tested here.

refs: https://issues.folio.org/browse/UISP-11

### CODE REVIEW

**Comment by K-Felk** [comment]
i figured out the linter weirdness-it looks like there was a webpac subdirectory in the main test directory that was full of cruft and was not needed to run tests-I suspect this was placed by the "stripes app bigTest" command.  Deleting the directory makes linter happy and the tests run fine without it.  Added a note to the bug I filed asking whoever fixes this command to check on it.

File: test/bigtest/network/boot.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import startMirage from './start';
```
Comment (mkuklis) [inline test/bigtest/network/boot.js:1]: @K-Felk could we do something similar to:

https://github.com/folio-org/ui-users/blob/master/test/bigtest/network/boot.js

here?

That way the `start.js` file could be also removed from this PR.

File: test/bigtest/network/start.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/* eslint global-require: off, import/no-mutable-exports: off */
```
Comment (mkuklis) [inline test/bigtest/network/start.js:1]: @K-Felk please see my comment above. I think we should be able to get rid of this file.

File: package.json [hunk 3]
```diff
@@ -15,8 +15,10 @@
     "actsAs": [
       "handler"
     ],
+    "stripes.type": "app",
```
Comment (zburke) [inline package.json:18]: I don't think you want the `stripes.` prefix here; that's implied by being a property of the `stripes` object starting on line 14.

File: package.json [hunk 4]
```diff
@@ -30,26 +32,45 @@
...
+    "react-router": "^5.2.0",
+    "react-router-dom": "^5.2.0",
+    "sinon": "^7.2.2"
   },
   "dependencies": {
+    "@folio/stripes-components": "~8.0.0",
+    "@folio/stripes-connect": "~6.0.0",
+    "core-js": "^3.6.4",
+    "@folio/stripes": "^4.0.0",
+    "@folio/stripes-core": "^5.0.0",
```
Comment (zburke) [inline package.json:62]: Not quite right; the deps should include `@folio/stripes` only. If you need access to un-exported things from, e.g. stripes-components for interactors in testing, then you can include those packages as dev-deps. At the dependencies level, however, the only stripes import should be `@folio/stripes`.

File: package.json [hunk 5]
```diff
@@ -30,26 +32,45 @@
   "scripts": {
     "start": "stripes serve",
     "lint": "eslint .",
-    "test": "echo 'placeholder. no tests implemented'"
+    "test": "stripes test karma"
   },
   "devDependencies": {
-    "@folio/eslint-config-stripes": "^5.1.0",
-    "@folio/stripes": "^4.0.0",
     "@folio/stripes-cli": "^1.3.1",
```
Comment (zburke) [inline package.json:38]: I saw @mkuklis [add support for `type: "handler"`](https://github.com/folio-org/stripes-cli/pull/231) in stripes-cli so we should use that value above and depend on a compatible version of `"@folio/stripes-cli": "^1.18.0"` here.

File: package.json (hunk 3 above)
Comment (zburke) [reply package.json:18]: Also, I saw @mkuklis [add support for `type: "handler"`](https://github.com/folio-org/stripes-cli/pull/231) in stripes-cli so we can use that here, right?

File: test/bigtest/network/boot.js (hunk 1 above)
Comment (K-Felk) [reply test/bigtest/network/boot.js:1]: Works fine, done, and removing start.js

File: package.json (hunk 4 above)
Comment (zburke) [reply package.json:62]: Also, if you're depending on react-router v5 then your stripes dep should be `^5.0.0`. In fact, the components and core deps you have here are incompatible with each other, which is exactly why we created `@folio/stripes` as a single entity that handles aligning all those deps for you.

**Comment by K-Felk** [comment]
I'm going to hold off on merging this until my stripes-cli updates with the change Michael pushed. The second test (which tests the servicepoint selection modal) sorta depends on ui-sp being mounted as an app with an endpoint you can visit-I want to make sure that having the testing suite treat it as a handler alone will not bork that test.  If it does, I'll have to rewrite it to use a harness.

**Comment by K-Felk** [comment]
*sigh* As suspected, the second set of tests is now borked because the module isn't being loaded as an app.  I'll have to rewrite them.  Closing this for now.

**Pull request closed without merging**